core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1127,6 +1138,81 @@ public String toString() {
                     "uid=" + uid +
                     '}';
         }
+
+    }
+
+    public static class NoOp implements Operation {
+
+        private final long seqNo;
+        private final long primaryTerm;
+        private final String reason;
+
+        public long seqNo() {
+            return seqNo;
+        }
+
+        public long primaryTerm() {
+            return primaryTerm;
+        }
+
+        public String reason() {
+            return reason;
+        }
+
+        NoOp(final StreamInput in) throws IOException {
+            seqNo = in.readVLong();
+            primaryTerm = in.readVLong();
+            reason = in.readString();
+        }
+
+        public NoOp(final long seqNo, final long primaryTerm, final String reason) {
+            assert seqNo > SequenceNumbersService.NO_OPS_PERFORMED;
+            assert primaryTerm >= 0;
+            assert reason != null;
+            this.seqNo = seqNo;
+            this.primaryTerm = primaryTerm;
+            this.reason = reason;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeVLong(seqNo);
+            out.writeVLong(primaryTerm);
+            out.writeString(reason);
+        }
+
+        @Override
+        public Type opType() {
+            return Type.NO_OP;
+        }
+
+        @Override
+        public long estimateSize() {
+            return 2 * reason.length() + StreamOutput.lengthVLong(seqNo) + StreamOutput.lengthVLong(primaryTerm);
```

Comment:
same issue here - we can just use size of long here - reason.length() is also wrong here - we write as utf8.